### PR TITLE
net/gcoap: fix guards around defines

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

The guards were wrong, leading to compilation issues with the modules enabled.

### CODE REVIEW

File: sys/include/net/gcoap.h [hunk 1]
```diff
@@ -639,10 +639,12 @@ extern "C" {
 #define GCOAP_VFS_EXTRA_STACKSIZE   (0)
 #endif
 
+#ifndef GCOAP_STACK_SIZE
 #define GCOAP_STACK_SIZE (THREAD_STACKSIZE_DEFAULT + DEBUG_EXTRA_STACKSIZE \
                           + sizeof(coap_pkt_t) + GCOAP_DTLS_EXTRA_STACKSIZE \
                           + GCOAP_VFS_EXTRA_STACKSIZE)
 #endif
+#endif
```
Comment (mguetschow) [inline sys/include/net/gcoap.h:647]: Hum, so now `GCOAP_STACK_SIZE` is nested within `#ifndef GCOAP_DTLS_EXTRA_STACKSIZE`, looks suspicious. If intended, would need an explanatory comment and maybe a closing `#endif` comment

**Review by mguetschow — CHANGES_REQUESTED** [review]
Looked straightforward on the first glance, but after a closer look I'm confused about the nesting here.

File: sys/include/net/gcoap.h (hunk 1 above)
Comment (benpicco) [reply sys/include/net/gcoap.h]: uh, good catch

**Review by mguetschow — APPROVED** [review]
Looks good now, thanks!

**Pull request merged**